mlir-sycl/lib/Dialect/IR/SYCLOps.cpp

```diff
@@ -92,15 +92,28 @@ bool SYCLAddrSpaceCastOp::areCastCompatible(TypeRange inputs,
 LogicalResult SYCLAccessorGetPointerOp::verify() {
   const auto accTy = cast<AccessorType>(
       cast<MemRefType>(getOperand().getType()).getElementType());
-  const Type resTy = getResult().getType();
-  const Type resElemTy = cast<MemRefType>(resTy).getElementType();
+  const MemRefType resTy = getResult().getType();
+  const Type resElemTy = resTy.getElementType();
   return (resElemTy != accTy.getType())
              ? emitOpError(
                    "Expecting a reference to this accessor's value type (")
                    << accTy.getType() << "). Got " << resTy
              : success();
 }
 
+LogicalResult SYCLAccessorGetRangeOp::verify() {
+  const auto accTy = cast<AccessorType>(
+      cast<MemRefType>(getOperand().getType()).getElementType());
+  const auto accRangeTy = cast<RangeType>(
+      cast<AccessorImplDeviceType>(accTy.getBody()[0]).getBody()[1]);
+  const RangeType resTy = getResult().getType();
+  return (accRangeTy != resTy)
+             ? emitOpError(
+                   "Expecting a reference to this accessor's range type (")
+                   << accRangeTy << "). Got " << resTy
+             : success();
```

Comment:
I don't think we should check the body to verify the operation (DPC++ specific). Can we simply compare the dimension numbers instead?